client/blocks/comments/post-comment-list.jsx

```diff
@@ -55,26 +71,81 @@ class PostCommentList extends React.Component {
+		props.startingCommentId &&
+		props.commentsTree[ this.props.startingCommentId ] &&
+		props.commentsFetchingStatus.hasReceivedBefore &&
+		props.commentsFetchingStatus.hasReceivedAfter &&
+		! this.hasScrolledToComment &&
+		window.document.getElementsByName( `comment-${ props.startingCommentId }` ).length > 0;
+
+	shouldFetchInitialComment = ( { startingCommentId, commentsTree } ) =>
+		startingCommentId && commentsTree && ! commentsTree[ startingCommentId ];
+
+	shouldFetchInitialPages = ( { commentsTree, startingCommentId } ) =>
+		startingCommentId &&
+		commentsTree[ startingCommentId ] &&
+		this.props.commentsTree[ startingCommentId ] &&
+		! this.alreadyLoadedInitialSet;
 
-		this.props.requestPostComments( siteId, postId, this.props.commentsFilter );
+	componentWillMount() {
+		const { post: { ID: postId, site_ID: siteId }, commentsFilter: status } = this.props;
+
+		if ( this.shouldFetchInitialComment( this.props ) ) {
+			this.props.requestComment( { siteId, commentId: this.props.startingCommentId } );
+		} else if ( this.shouldFetchInitialPages( this.props ) ) {
+			this.viewEarlierCommentsHandler();
+			this.viewLaterCommentsHandler();
+			this.alreadyLoadedInitialSet = true;
+		} else {
+			this.props.requestPostComments( { siteId, postId, status } );
+		}
 	}
 
 	componentWillReceiveProps( nextProps ) {
 		const nextSiteId = get( nextProps, 'post.site_ID' );
 		const nextPostId = get( nextProps, 'post.ID' );
 		const nextCommentsFilter = get( nextProps, 'commentsFilter' );
-
-		if (
-			nextSiteId &&
-			nextPostId &&
-			nextCommentsFilter &&
-			( this.props.post.site_ID !== nextSiteId ||
-				this.props.post.ID !== nextPostId ||
-				this.props.commentsFilter !== nextCommentsFilter )
+		const nextInitialComment = nextProps.initialComment;
+
+		if ( this.shouldFetchInitialComment( nextProps ) && ! nextProps.initialComment ) {
+			this.props.requestComment( { siteId: nextSiteId, commentId: nextProps.startingCommentId } );
+			this.hasScrolledToComment = false;
+		} else if ( this.shouldFetchInitialPages( nextProps ) ) {
+			this.viewEarlierCommentsHandler();
+			this.viewLaterCommentsHandler();
+			this.alreadyLoadedInitialSet = true;
+		} else if (
+			( nextSiteId &&
+				nextPostId &&
+				nextCommentsFilter &&
+				( this.props.post.site_ID !== nextSiteId ||
+					this.props.post.ID !== nextPostId ||
+					this.props.commentsFilter !== nextCommentsFilter ) ) ||
+			( this.props.initialComment !== nextInitialComment &&
+				( ( nextInitialComment.post && nextInitialComment.post.ID !== nextPostId ) ||
+					nextInitialComment.error ) )
 		) {
-			this.props.requestPostComments( nextSiteId, nextPostId, this.props.commentsFilter );
+			nextProps.requestPostComments( {
+				siteId: nextSiteId,
+				postId: nextPostId,
+				status: nextCommentsFilter,
+			} );
+		}
+
+		if ( this.shouldScrollToComment( nextProps ) ) {
+			// defer so that the above/below comments have time to render
+			defer( () => this.scrollToComment(), 50 );
```

Comment:
`defer` doesn't take a timeout value. it passes args on to the callback instead. do you mean to pass 50 or are you trying to wait 50ms?